applications/tari_testnet_miner/src/main.rs

```diff
@@ -100,6 +113,24 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
     Ok(())
 }
 
+async fn mine(miner: &mut Miner, header: BlockHeader) {
+    miner.mine(header);
+}
+
+async fn stop_mining() {
+    loop {
+        let mut line = String::new();
+        println!("Mining, press c to close and stop");
+        let b1 = std::io::stdin().read_line(&mut line).unwrap();
```

Comment:
This is blocking and so will break the scheduler.

I think something like this will work:

```rust
let stdin_reader = BufReader::new(std::io::stdin());
let buf = Vec::new();

let (_, buf) = io.read_until(stdin_reader, b'\n', buf).await;
if buf == b"c" {
   break;
}
```

Relevant docs: 
https://docs.rs/tokio-io/0.2.0-alpha.5/i686-apple-darwin/tokio_io/struct.BufReader.html
https://docs.rs/tokio-io/0.2.0-alpha.6/tokio_io/trait.AsyncBufReadExt.html#method.read_until